### PR TITLE
Add empty smithy-event-stream package

### PR DESCRIPTION
**Pull request by JordonPhillips**
This adds an empty package to contain the event stream implementation.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python-packages/smithy-event-stream/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+[build-system]
+requires = ["setuptools", "setuptools-scm", "wheel"]
+build-backend = "setuptools.build_meta"
```
Comment (nateprewitt) [inline python-packages/smithy-event-stream/pyproject.toml:3]: Are we using setuptools in the other projects? Not part of this PR but as a note, we should definitely move to [`hatchling`](https://hatch.pypa.io/latest/) over setuptools.

**Review by nateprewitt — APPROVED** [review]

File: python-packages/smithy-event-stream/pyproject.toml (hunk 1 above)
Comment (JordonPhillips) [reply python-packages/smithy-event-stream/pyproject.toml:3]: We are - but we can switch. I have no strong opinions on the topic. Lets do it as one big changeover though
